### ISSUE TITLE
Chunked writing in the server

### ISSUE THREAD

**Issue by lodejard** [aspnet/KestrelHttpServer#615]
Consider adding a bool argument to the internal methods where writes are being passed down.

Down at the layer where the blit into the buffer pool memory is taking place, reserve a slightly larger range of data, and put the chunk leaders/trailer around the data in that single write.

I'm not sure if the system already does this :) just filing a bug to verify

**Comment by halter73** [comment]
We flow the bool chunk argument a little deeper than we did a few months ago, but it could still possibly be better optimized.

We still create a chunk per the user's call to write instead of delaying it until we call uv_write. Doing the latter would be somewhat difficult though, since we would need to keep track of the beginning and end of the automatically chunked response body at a low level.

I also experimented with an [allocation free ChunkWriter](https://github.com/aspnet/KestrelHttpServer/commit/d94d075bab4ad8d8156209f3e550ab53b5831da1). I did some wrk benchmarks on large chunked responses though and it didn't really make a measurable improvement over the current version where we allocate a 10 byte array per chunk.
